### PR TITLE
통합 AutoNN : YoloE 기능 통합 & 관련 상태 업데이트 POST 기능 추가 

### PR DESCRIPTION
**Pull request by HyunwooCho**
### PR 타입
하나 이상의 PR 타입을 선택해주세요.
- [X] 기능 추가
- [ ] 기능 삭제
- [ ] 버그 수정
- [X] 의존성, 환경 변수, 빌드 관련 코드 업데이트
- [ ] 기타

### PR전 체크 사항
PR전 아래 사항을 확인하였는지 체크하시고 PR 생성 바랍니다.
- [X] 자신의 브랜치로 main 브랜치 최신 내용 반영후 동작 테스트 완료
- [X] 다른 작업자의 폴더내 파일 수정하지 않음.
- [X] docker-compose.yml 수정한 경우, 수정에 따른 영향 평가 

### 관련 이슈
#157
#162
#163
#165

### PR 반영 브랜치
ETRI-tenace-AutoNN -> main

### 변경 사항
* YoloE (YOLOv7) 기능 통합
* 관련 status_update POST 구현 완료
* Checkpoint (last.pt) 및 관련 파일 공유 볼륨에 저장
   * /shared/common/userid/projectid/exp/weights/last.pt

### 테스트 결과
* PM에서 AutoNN의 GET /start 호출 **<AutoNN 시작>**
* AutoNN에서 PM으로 기본 셋팅 POST /status_update 호출 (1초 이내)
  * project_info
  * hyperparameter
  * arguments
  * basemodel
  * system
  * model
  * model_summary

* AutoNN에서 배치 사이즈 계산 POST /status_update 반복 호출 (약 2분 소요) 
  * batchsize (신규 생성)

* AutoNN에서 환경변수 등을 업데이트한 arguments 다시 POST
  * arguments
* AutoNN에서 데이터셋 체크
  * train_dataset (반복/5000개마다 한 번씩)
  * val_dataset (반복/5000개마다 한 번씩)
* AutoNN 학습 (COCO 데이터셋 기준 train 루틴 17분 정도 소요)
  * train_start
  * hyperparameter (매 epoch 마다 learning rate/momentum 갱신)
  * train_loss (매 epoch에 대해 배치 사이즈 step 마다 한 번씩 보냄)
  * val_accuracy (매 epoch에 대해 한 번만 보냄)
  * val_accuracy (마지막 한 번 더 보냄; 최종 결과)
  * train_end
* AutoNN에서 GET /status_report completed 호출 **<AutoNN 끝>**

### CODE REVIEW

**Review by ML-TANGO — APPROVED** [review]
PR 요청 내용 확인 및 승인

**Comment by ML-TANGO** [comment]
휴지통 아이콘으로 프로젝트 삭제후, 새로운 프로젝트 생성하여 파이프라인 시작시 start API에 잘못된 project_id가 사용되는 경우 발생
* 휴지통 아이콘으로 프로젝트 삭제시, /shared/common/user_id/project_id 폴더도 삭제해야 함.
